lib/Util/Builtin/CompilerStdFlags.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021 Nicholas Corgan
+// SPDX-License-Identifier: BSL-1.0
+
+#include <Poco/Platform.h>
+
+// Only supported in MSVC 2017+ (no C++11 flag)
+#if defined(POCO_COMPILER_MSVC) && _MSC_VER >= 1910
+    #define CPP11_COMPILER_FLAG ""
+    #define CPP14_COMPILER_FLAG "/std:c++14"
+    #define CPP17_COMPILER_FLAG "/std:c++17"
+    #define CPP20_COMPILER_FLAG "/std:c++latest"
+#elif defined(POCO_COMPILER_GCC) || defined(POCO_COMPILER_MINGW) || defined(POCO_COMPILER_CLANG)
+    #define CPP11_COMPILER_FLAG "-std=c++11"
+    #define CPP14_COMPILER_FLAG "-std=c++14"
+    #define CPP17_COMPILER_FLAG "-std=c++17"
+    #define CPP20_COMPILER_FLAG "-std=c++20"
+#else
+    #define CPP11_COMPILER_FLAG ""
+    #define CPP14_COMPILER_FLAG ""
+    #define CPP17_COMPILER_FLAG ""
+    #define CPP20_COMPILER_FLAG ""
+#endif
+
+#ifdef POCO_COMPILER_MSVC
+#define CPP_STD _MSVC_LANG
+#else
```

Comment:
https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/

> The MSVC compiler’s definition of the __cplusplus predefined macro leaps ahead 20 years in Visual Studio 2017 version 15.7 Preview 3.
> You need to compile with the /Zc:__cplusplus switch to see the updated value of the __cplusplus macro. We tried updating the macro by default and discovered that a lot of code doesn’t compile correctly

lol nice

I think I can set the `/Zc:__cplusplus` flag as part of the cmake work so we can just use it everywhere